### PR TITLE
Fix dc example

### PR DESCRIPTION
**Pull request by luisacicolini**
Function `splitOn` is defined using `splitOnAux`, which is defined by well-founded recursion and marked @[irreducible] by default. As [advised](https://github.com/leanprover/lean4/pull/4061), we set `splitOnAux` back to semireducible by using `unseal` to use it in `DCExample`. CC: @ymherklotz @alexkeizer

### CODE REVIEW

File: SSA/Projects/CIRCT/DCExample.lean [hunk 1]
```diff
@@ -30,16 +28,15 @@ def BranchEg1 := [dc_com| {
...
-def test : Stream Bool :=
-  BranchEg1.denote (Valuation.ofPair c x)
+def test : Stream Int :=
+  BranchEg1.denote (Ctxt.Valuation.ofPair c x)
 
 def remNone (lst : List (Option Bool)) : List (Option Bool) :=
   lst.filter (fun | some x => true
                   | none => false)
 
-
```
Comment (tobiasgrosser) [inline SSA/Projects/CIRCT/DCExample.lean:43]: This is unrelated. Maybe not include this in this commit.

File: SSA/Projects/CIRCT/DCExample.lean [hunk 2]
```diff
@@ -3,24 +3,22 @@ import SSA.Core.MLIRSyntax.GenericParser
...
-    "return" (%out) : (!Stream_Bool) -> ()
+  ^entry(%0: !Stream_Int, %1: !Stream_Bool):
+    %out = "dc.branch" (%0, %1) : (!Stream_Int, !Stream_Bool) -> (!Stream2_Int)
+    %outf = "dc.fst" (%out) : (!Stream2_Int) -> (!Stream_Int)
+    %outs = "dc.snd" (%out) : (!Stream2_Int) -> (!Stream_Int)
+    %out2 = "dc.merge" (%outs, %outf) : (!Stream_Int, !Stream_Int) -> (!Stream_Int)
+    "return" (%out2) : (!Stream_Int) -> ()
   }]
 
-
```
Comment (tobiasgrosser) [inline SSA/Projects/CIRCT/DCExample.lean:23]: This is unrelated. Maybe not include this in this commit.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -63,6 +63,8 @@ jobs:
         run: |
           lake -R exe cache get # download cache of mathlib docs.
           lake -R build SSA.Projects.CIRCT.DC # compile and check CIRCT's DC Dialect
+          lake -R build SSA.Projects.CIRCT.DCExample # test CIRCT's DC Dialect
+
```
Comment (tobiasgrosser) [inline .github/workflows/ci.yml]:
```suggestion
          lake -R build SSA.Projects.CIRCT.DCExample # test CIRCT's DC Dialect

```

File: SSA/Projects/CIRCT/DC.lean [hunk 4]
```diff
@@ -208,7 +208,6 @@ def mkTy2 : String → MLIR.AST.ExceptM (DC) Ty2
   | "Bool" => return (.bool)
   | _ => throw .unsupportedType
 
-
```
Comment (tobiasgrosser) [inline SSA/Projects/CIRCT/DC.lean:211]: This is unrelated.

**Review by tobiasgrosser — APPROVED** [review]
How nice. Thank you for debugging this.

**Comment by tobiasgrosser** [comment]
Also, you may want to update the commit message to include @ymherklotz's analysis.

File: SSA/Projects/CIRCT/DCExample.lean [hunk 5]
```diff
@@ -3,24 +3,22 @@ import SSA.Core.MLIRSyntax.GenericParser
 import SSA.Projects.CIRCT.DC
 import SSA.Projects.CIRCT.DC.Stream
 
-
-
```
Comment (tobiasgrosser) [inline SSA/Projects/CIRCT/DCExample.lean:7]: Unrelated.

**Review by ymherklotz — APPROVED** [review]
This looks good to me! I think `unseal` is fine for now, but maybe there is a way to handle it a bit better in the framework in the future.
